addons/mail/controllers/discuss/channel.py

```diff
@@ -57,11 +57,9 @@ def discuss_channel_messages(self, channel_id, search_term=None, before=None, af
             ("model", "=", "discuss.channel"),
             ("message_type", "!=", "user_notification"),
         ]
-        res = request.env["mail.message"]._message_fetch(
-            domain, search_term=search_term, before=before, after=after, around=around, limit=limit
-        )
+        res = request.env["mail.message"]._message_fetch(domain, **(fetch_params or {}))
         messages = res.pop("messages")
-        if not request.env.user._is_public() and not around:
+        if not request.env.user._is_public() and "around" not in (fetch_params or {}):
```

Comment:
Technically this is still not the same as before. Checking dict existence is ignoring whether the value is true/false, if you want to use the "or {}" thing, you'd need to use get() on the key instead.